plugins/UpdateChecker/UpdateChecker.py

```diff
@@ -9,54 +9,35 @@
 from UM.Message import Message
 from UM.Version import Version
 from UM.Preferences import Preferences
+from UM.Job import Job
 
 import urllib.request
 import platform
 import json
 import codecs
 import webbrowser
-from threading import Thread
 
 i18n_catalog = i18nCatalog("uranium")
 
-## This extention checks for new versions of the application based on the application name and the version number.
-#  The plugin is currently only usuable for applications maintained by Ultimaker. But it should be relatively easy
-#  to change it to work for other applications.
-class UpdateChecker(Extension):
-    def __init__(self):
+class UpdateCheckerJob(Job):
+    def __init__(self, silent = False, url = None):
         super().__init__()
-        self.addMenuItem(i18n_catalog.i18nc("@item:inmenu", "Check for Updates"), self.checkNewVersion)
-        self._url = None
-
-        Preferences.getInstance().addPreference("info/automatic_update_check", True)
-        if Preferences.getInstance().getValue("info/automatic_update_check"):
-            thread = Thread(target = self.checkNewVersion, args = (True,))
-            thread.daemon = True
-            thread.start()
-
-    ##  Callback for the message that is spawned when there is a new version.
-    def actionTriggered(self, message, action):
-        if action == "download":
-            if self._url is not None:
-                webbrowser.open(self._url)
-
-    ##  Connect with software.ultimaker.com, load latest.json and check version info.
-    #   If the version info is higher then the current version, spawn a message to
-    #   allow the user to download it.
-    #
-    #   \param silent type(boolean) Suppresses messages other than "new version found" messages.
-    #                               This is used when checking for a new version at startup.
-    def checkNewVersion(self, silent = False):
+        self.silent = silent
+        self.url = url
+    
+    def run(self):
+        if not self.url:
+            Logger.log("e", "Can not check for a new release. URL not set!")
         no_new_version = True
 
         application_name = Application.getInstance().getApplicationName()
         Logger.log("i", "Checking for new version of %s" % application_name)
 
         try:
-            latest_version_file = urllib.request.urlopen("http://software.ultimaker.com/latest.json")
-        except Exception as e:
-            Logger.log("e", "Failed to check for new version. %s" %e)
-            if not silent:
+            latest_version_file = urllib.request.urlopen(self.url)
+        except Exception:
+            Logger.logException("e", "Failed to check for new version.")
```

Comment:
Logger will do this for us. When looking at logException() it says that you only need to pass an message and the last message of an exception will be taken from the traceback module.
See: https://github.com/Ultimaker/Uranium/blob/master/UM/Logger.py#L53

So in my point of view no need to take the exception and pass it to the Logger.